### PR TITLE
Increases test coverage for server module

### PR DESCRIPTION
**Pull request by se7entyse7en**
Increases test coverage for server module: 66.3% -> 73.9%.

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -42,158 +44,203 @@ var correctReviewEvent = lookout.ReviewEvent{
...
+
+	// send other event without timeout
+	err = watcher.Send(pushEvent)
+	s.Nil(err)
+
+	comments = poster.PopComments()
+	s.Len(comments, 1)
+}
+
+func (s *ServerTestSuite) TestPushTimeout() {
```
[the remark this answers is not in the document]
Comment (se7entyse7en) [reply server/server_test.go]: Prefer to keep them separated to make the test more readable.

File: server/server_test.go [hunk 2]
```diff
@@ -42,158 +44,203 @@ var correctReviewEvent = lookout.ReviewEvent{
...
+	s.Equal(makeComment(pushEvent.CommitRevision.Base, pushEvent.CommitRevision.Head), comments[0])
 
 	status := poster.PopStatus()
-	require.Equal(lookout.SuccessAnalysisStatus, status)
+	s.Equal(lookout.SuccessAnalysisStatus, status)
 }
 
-func TestServerPersistedReview(t *testing.T) {
-	require := require.New(t)
+func (s *ServerTestSuite) TestReviewTimeout() {
```
[the remark this answers is not in the document]
Comment (se7entyse7en) [reply server/server_test.go]: Same as below.

File: server/server_test.go [hunk 3]
```diff
@@ -42,158 +44,203 @@ var correctReviewEvent = lookout.ReviewEvent{
...
 }
 
-func TestServerReview(t *testing.T) {
-	require := require.New(t)
+type ServerTestSuite struct {
+	suite.Suite
+}
 
-	watcher, poster := setupMockedServer()
+func (s *ServerTestSuite) TestReview() {
```
[the remark this answers is not in the document]
Comment (se7entyse7en) [reply server/server_test.go]: Same as below.

File: server/server_test.go [hunk 4]
```diff
@@ -42,158 +44,203 @@ var correctReviewEvent = lookout.ReviewEvent{
...
+	s.Equal(makeComment(reviewEvent.CommitRevision.Base, reviewEvent.CommitRevision.Head), comments[0])
 
 	status := poster.PopStatus()
-	require.Equal(lookout.SuccessAnalysisStatus, status)
+	s.Equal(lookout.SuccessAnalysisStatus, status)
 }
 
-func TestServerPush(t *testing.T) {
-	require := require.New(t)
+func (s *ServerTestSuite) TestPush() {
```
[the remark this answers is not in the document]
Comment (se7entyse7en) [reply server/server_test.go]: Same as below.

**Comment by carlosms** [comment]
CI is failing for the web tests.

File: server/server_test.go [hunk 5]
```diff
@@ -42,158 +44,202 @@ var correctReviewEvent = lookout.ReviewEvent{
...
 
 	err := watcher.Send(reviewEvent)
-	require.Nil(err)
+	s.Nil(err)
 
 	comments := poster.PopComments()
-	require.Len(comments, 1)
-	require.Equal(makeComment(reviewEvent.CommitRevision.Base, reviewEvent.CommitRevision.Head), comments[0])
+	s.Len(comments, 1)
+	s.Equal(makeComment(reviewEvent.CommitRevision.Base, reviewEvent.CommitRevision.Head), comments[0])
```
[the remark this answers is not in the document]
Comment (se7entyse7en) [reply server/server_test.go]: Ok! Thanks for the explanation!

[the remark this answers is not in the document]
Comment (se7entyse7en) [reply server/server_test.go]: done

**Comment by se7entyse7en** [comment]
Rebased, waiting for CI before merge

File: server/server_test.go [hunk 6]
```diff
@@ -98,21 +111,80 @@ func TestServerPush(t *testing.T) {
...
+	require.Nil(err)
+
+	comments = poster.PopComments()
+	require.Len(comments, 1)
+}
+
+func (s *ServerTestSuite) TestPushTimeout() {
+	require := s.Require()
+
+	pushSleep, _ := time.ParseDuration("5ms")
```
Comment (smacker) [inline server/server_test.go]: usually, it's written as:

```go
5 * time.Millisecond
```

**Pull request merged**